### PR TITLE
Define who will pay the expense fees (Transferwise only)

### PR DESCRIPTION
**Pull request by Betree**
MVP for https://github.com/opencollective/opencollective/issues/1073

When paid by the payee, fees are stored in `data.payeeFees`. We may one day want to store these as separate transactions.

### CODE REVIEW

**Comment by abitrolly** [comment]
Is there an API call to get the transaction with the info who paid the fee, how much was predicted, and how much was actually paid?

File: server/graphql/common/expenses.ts [hunk 1]
```diff
@@ -1244,16 +1298,15 @@ export async function payExpense(req: express.Request, args: Record<string, unkn
...
+        } else if (forceManual) {
           await createTransactions(host, expense, feesInHostCurrency);
         } else if (paypalPaymentMethod) {
           return payExpenseWithPayPal(remoteUser, expense, host, paypalPaymentMethod, paypalEmail, feesInHostCurrency);
         } else {
           throw new Error('No Paypal account linked, please reconnect Paypal or pay manually');
         }
       } else if (payoutMethodType === PayoutMethodTypes.BANK_ACCOUNT) {
-        if (args.forceManual) {
-          feesInHostCurrency['paymentProcessorFeeInHostCurrency'] = 0;
```
Comment (Betree) [inline server/graphql/common/expenses.ts:1256]: Removing this line fixes an issue introduced 2 years ago: we've been stripping out manually set payment processor fees.

**Comment by Betree** [comment]
> Is there an API call to get the transaction with the info who paid the fee, how much was predicted, and how much was actually paid?

This PR introduces the `expense.feesPayer` field in that regard. All existing expenses have it set to `COLLECTIVE`, because that's how things have been working until now. When paid by the collective, fields can be retrieved from the public GraphQL API (`transaction.paymentProcessorFee`). Fees paid by the payee are not exposed yet.

**Review by znarf — APPROVED** [review]

**Comment by znarf** [comment]
Looks good.

File: server/graphql/common/expenses.ts [hunk 2]
```diff
@@ -980,13 +977,21 @@ async function createTransactions(
   data = {},
 ) {
   debug('marking expense as paid and creating transactions in the ledger', expense.id);
+  let paymentProcessorFeeInHostCurrency = fees['paymentProcessorFeeInHostCurrency'] || 0;
+
+  // Move the fee to `data` if paid by the payee
+  if (expense.feesPayer === 'PAYEE') {
+    data['payeeFees'] = { paymentProcessorFeeInHostCurrency };
+    paymentProcessorFeeInHostCurrency = 0;
```
Comment (Betree) [inline server/graphql/common/expenses.ts:985]: With the upcoming changes in #6899, we need to record these in the ledger

**Review by znarf — APPROVED** [review]

**Pull request merged**